### ISSUE TITLE
Throw Failed installation exception if the installation failed due to storage issues

### ISSUE THREAD

**Issue by DetroitLibertyPenguin** [Tobi823/ffupdater#300]
Stacktrace:
```
java.lang.RuntimeException: Failed to install FENNEC_FDROID in the background.
	at de.marmaro.krt.ffupdater.BackgroundJob$installApplication$3.invokeSuspend(SourceFile:142)
	at z3.a.resumeWith(Unknown Source:8)
	at m4.i0.run(Unknown Source:100)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8751)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
Caused by: de.marmaro.krt.ffupdater.installer.exception.InstallationFailedException: The installation failed because not enough storage is available.
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.fail(Unknown Source:4)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.install(Unknown Source:83)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.access$install(Unknown Source:0)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller$executeInstallerSpecificLogic$3.invokeSuspend(Unknown Source:37)
	... 10 more
Caused by: java.io.IOException: Requested internal only, but not enough space
	at java.lang.reflect.Constructor.newInstance0(Native Method)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
	at android.os.ParcelableException.readFromParcel(ParcelableException.java:56)
	at android.os.ParcelableException$1.createFromParcel(ParcelableException.java:82)
	at android.os.ParcelableException$1.createFromParcel(ParcelableException.java:79)
	at android.os.Parcel.readParcelable(Parcel.java:3346)
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2433)
	at android.os.Parcel.createException(Parcel.java:2422)
	at android.os.Parcel.readException(Parcel.java:2405)
	at android.os.Parcel.readException(Parcel.java:2347)
	at android.content.pm.IPackageInstaller$Stub$Proxy.createSession(IPackageInstaller.java:548)
	at android.content.pm.PackageInstaller.createSession(PackageInstaller.java:446)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.install(Unknown Source:17)
	... 12 more
Caused by: android.os.RemoteException: Remote stack trace:
	at android.util.ExceptionUtils.wrap(ExceptionUtils.java:34)
	at com.android.server.pm.PackageInstallerService.createSession(PackageInstallerService.java:643)
	at android.content.pm.IPackageInstaller$Stub.onTransact(IPackageInstaller.java:235)
	at android.os.Binder.execTransactInternal(Binder.java:1215)
	at android.os.Binder.execTransact(Binder.java:1179)
```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.3.5 (119) release |
| Device | SM-S515DL (a51tfn, a51, exynos9611) |
| Manufacturer | samsung (samsung) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 12 (SDK: 31) |
| OS | 21DJFB03, dpi, release-keys, 1664275168000 |

**Comment by Tobi823** [comment]
> The installation failed because not enough storage is available.

Your smartphone does not have enough memory/storage to install the update. Try to free up some storage by deleting unused apps or documents.

Or do i missed something?

**Comment by DetroitLibertyPenguin** [comment]
Yeah, all I was saying is that I wouldn't expect the app to crash in that instance, that it would check and give me a failure message only.

**Comment by Tobi823** [comment]
I thought that you just copied the detailed error message and that the app did not crash.

FFUpdater already checks for enough storage - but the check is complex and not perfect.

I see the problem - I have to throw a InstallationFailedException to show the error in the UI without crashing the app.

**Comment by Tobi823** [comment]
Note to me: maybe it is already fixed in 77.6.3?
